### PR TITLE
Add initialize SDL task

### PR DESCRIPTION
**Pull request by jonfortescue**
Adds a task to initialize SDL tools. SDL tool settings are pulled from [this internal repository](https://dev.azure.com/dnceng/internal/_git/sdl-tool-cfg?path=%2F&version=GBmaster).

### CODE REVIEW

File: src/Microsoft.DotNet.SDLAutomator/Microsoft.DotNet.SDLAutomator.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SDLAutomator/Microsoft.DotNet.SDLAutomator.csproj]: Does the project really need to be an SDK? Just curious about this.

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs [hunk 2]
```diff
@@ -6,20 +6,23 @@
 
 namespace Microsoft.DotNet.SDLAutomator
 {
-    public class InstallGuardian : MSBuild.Task
+    public class InstallSDL : MSBuild.Task
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: License header is missing in this file.

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs [hunk 3]
```diff
@@ -6,20 +6,23 @@
...
-        public string GuardianPackageVersion { get; set; }
+        public string SdlToolName { get; set; }
+
+        [Required]
+        public string SdlPackageVersion { get; set; }
 
         public override bool Execute()
         {
-            Log.LogMessage(MessageImportance.Low, "Inside InstallGuardian");
+            Log.LogMessage(MessageImportance.Low, "Inside InstallSDL");
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: I think low is the default for importance

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs [hunk 4]
```diff
@@ -28,13 +31,13 @@ public override bool Execute()
...
 
-        public void AddGuardianToPath()
+        public void AddSdlToolsToPath()
         {
             try
             {
                 const string name = "PATH";
                 string pathvar = System.Environment.GetEnvironmentVariable(name);
-                var value = $"{pathvar};{PackagesDirectory}\\microsoft.guardian.cli\\{GuardianPackageVersion}\\tools\\;";
+                var value = $"{pathvar};{PackagesDirectory}\\{SdlToolName}\\{SdlPackageVersion}\\tools\\;";
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: Can't you do some validations to check the path exist, etc?

Comment (JohnTortugo) [reply src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: Will this execute only on windows?

File: src/Microsoft.DotNet.SDLAutomator/InitSDL.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
+using System;
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.SDLAutomator/InitSDL.cs]: missing license header

File: src/Microsoft.DotNet.SDLAutomator/InitSDL.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using MSBuild = Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.SDLAutomator
+{
+    /// <summary>
+    /// This task should only ever be used in internal builds
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.SDLAutomator/InitSDL.cs]: Isn't it possible to add a check + warning for cases when someone try to use it on non-internal builds?

File: src/Microsoft.DotNet.SDLAutomator/build/Microsoft.DotNet.SDLAutomator.targets [hunk 7]
```diff
@@ -1,17 +1,21 @@
...
 
   <Target Name="Execute">
     <Error Text="The PackageDirectory property wasn't supplied."
-          Condition="'$(PackageDirectory)' == ''" />
+          Condition="'$(PackagesDirectory)' == ''" />
 
-    <InstallGuardian PackageDirectory="$(PackageDirectory)" 
-                     GuardianPackageVersion="0.0.38"/>
+    <InstallSDL PackagesDirectory="$(PackagesDirectory)"
+                     SdlToolName="$(SdlToolName)"
```
[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/build/Microsoft.DotNet.SDLAutomator.targets]: Should be fixed now.

File: src/Microsoft.DotNet.SDLAutomator/Microsoft.DotNet.SDLAutomator.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/Microsoft.DotNet.SDLAutomator.csproj]: It no longer is.

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs (hunk 4 above)
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: This no longer exists, so no worries.

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs (hunk 3 above)
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: No longer exists.

File: src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs (hunk 2 above)
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/InstallSDL.cs]: No longer exists.

File: src/Microsoft.DotNet.SDLAutomator/InitSDL.cs (hunk 6 above)
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/InitSDL.cs]: This is just a note -- I believe such a check already exists.

File: src/Microsoft.DotNet.SDLAutomator/InitSDL.cs (hunk 5 above)
Comment (jonfortescue) [reply src/Microsoft.DotNet.SDLAutomator/InitSDL.cs]: Will add!

**Pull request merged**